### PR TITLE
Documentation of the generic-server

### PR DESCRIPTION
**Pull request by ndefokou**
@francis-pouatcha @IngridPuppet @Hermann-Core

### CODE REVIEW

**Review by Hermann-Core — COMMENTED** [review]
There are typos in the documentation. Please review the file again and correct them.

**Comment by ndefokou** [comment]
@Hermann-Core typos corrected

**Review by Hermann-Core — COMMENTED** [review]
I suggest compacting the content a bit. Focus on the most important information and summarize where possible, when appropriate, add external links for further reading instead of including lengthy explanations. This will keep the documentation concise and provide additional resources for those who want to dive deeper.

**Review by IngridPuppet — CHANGES_REQUESTED** [review]
Check my review of #91. Your documentation seems more focused on what a generic server is in general, and this is not the right place to have such lecture. One is rather interested in what this generic server implementation has to offer.

**Comment by ndefokou** [comment]
@IngridPuppet @Hermann-Core, please check the improvement I did.

File: generic-server/README.md [hunk 1]
```diff
@@ -1,2 +1,34 @@
 # generic-server
-This server aggregates features provided by configurable plugins.
+
+This server provides a framework to build versatile applications by aggregating functionalities offered by configurable plugins.
+
```
[the remark this answers is not in the document]
Comment (ndefokou) [reply generic-server/README.md]: @Hermann-Core  Please review this again.

**Comment by ndefokou** [comment]
@Hermann-Core  Please review this again.

File: generic-server/README.md [hunk 2]
```diff
@@ -1,2 +1,51 @@
...
+- Define the necessary methods (e.g., name(), mount(), and routes()).
+
+**2-Register the Plugin:**
+
+- Add the plugin to the static PLUGINS array.
+
+**3-Load the Plugin:**
+
+- Ensure the plugin container is re-initialized to include the new plugin.**
```
Comment (IngridPuppet) [inline generic-server/README.md:23]: This step is not necessary, the plugins are loaded automatically as the server boots. I can't recall the plugin container being able to be re-initialized.

File: generic-server/README.md [hunk 3]
```diff
@@ -1,2 +1,51 @@
...
+## Extending Server Functionalities with Plugins
+### To add a new plugin:
+
+**1- Implement the `Plugin` Trait:**
+
+- Define the necessary methods (e.g., name(), mount(), and routes()).
+
+**2-Register the Plugin:**
+
+- Add the plugin to the static PLUGINS array.
```
Comment (IngridPuppet) [inline generic-server/README.md:19]: Improve the styling:
- Use conventional markdown numbered lists.
- There is no need to define a list when you only have one item. For example, I'm using a list to make suggestions here 'cause I have at least two of them.
- Preview your markdown code. You can use a live editor such as https://dillinger.io/.
<br>

```suggestion

**1. Implement the `Plugin` Trait:**

Define the necessary methods (e.g., name(), mount(), and routes()).

**2. Register the Plugin:**

Add the plugin to the static PLUGINS array.
```

Comment (IngridPuppet) [inline generic-server/README.md:19]: Indicate where to find that array.

**Review by IngridPuppet — CHANGES_REQUESTED** [review]

File: generic-server/README.md [hunk 4]
```diff
@@ -1,2 +1,47 @@
...
+## Extending Server Functionalities with Plugins
+### To add a new plugin:
+
+**1. Implement the `Plugin` Trait:**
+
+ Define the necessary methods (e.g., name(), mount(), and routes()).
+
+**2. Register the Plugin:**
+
+ Add the plugin to the static PLUGINS array.This array are located in your main server configuration file or a central location where the server initializes and manages plugins.
```
Comment (IngridPuppet) [inline generic-server/README.md:19]:
```suggestion
 Add the plugin to the static PLUGINS array (located in `src/plugin/mod.rs`).
```

**Review by IngridPuppet — CHANGES_REQUESTED** [review]

**Review by IngridPuppet — APPROVED** [review]

File: generic-server/README.md [hunk 5]
```diff
@@ -1,2 +1,47 @@
...
+    fn unmount(&self) -> Result<(), PluginError> {
+        // Initialization logic here
+        Ok(())
+    }
+    fn routes(&self) -> Router {
+        // Define and return routes here
+        Router::new().route("/myplugin", get(my_plugin_handler))
+    }
+}
+```
```
Comment (Hermann-Core) [inline generic-server/README.md:47]: Maybe you can also add another example here showing how the plugin can be added to the static `PLUGINS` array ?

**Review by Hermann-Core — CHANGES_REQUESTED** [review]

**Review by Hermann-Core — APPROVED** [review]

**Review by Hermann-Core — APPROVED** [review]

**Pull request merged**